File: charactermanager/backend/src/main/java/ai/ragu/charactermanager/enumeration/AttackProficiencyEnum.java
package ai.ragu.charactermanager.enumeration;

import com.fasterxml.jackson.annotation.JsonCreator;

public interface AttackProficiencyEnum {
    @JsonCreator
    static AttackProficiencyEnum create(String str) {
        try {
            return MeleeWeaponTypeEnum.fromValue(str);
        } catch (Exception ignored) {}
        try {
            return RangedWeaponTypeEnum.fromValue(str);
        } catch (Exception ignored) {}
        try {
            return OtherAttackWeaponTypeEnum.fromValue(str);
        } catch (Exception ignored) {}
        throw new IllegalArgumentException(String.format("Unknown attack proficiency: %s", str));
    }
}

File: charactermanager/backend/src/main/java/ai/ragu/charactermanager/enumeration/AttackProficiencyEnum.kt
package ai.ragu.charactermanager.enumeration

import com.fasterxml.jackson.annotation.JsonCreator

interface AttackProficiencyEnum {
    companion object {
        @JsonCreator
        fun create(str: String?): AttackProficiencyEnum? {
            try {
                return MeleeWeaponTypeEnum.Companion.fromValue(str)
            } catch (ignored: Exception) {
            }
            try {
                return RangedWeaponTypeEnum.Companion.fromValue(str)
            } catch (ignored: Exception) {
            }
            try {
                return OtherAttackWeaponTypeEnum.Companion.fromValue(str)
            } catch (ignored: Exception) {
            }
            throw IllegalArgumentException(String.format("Unknown attack proficiency: %s", str))
        }
    }
}